### PR TITLE
Typescript support!

### PR DESCRIPTION
**Pull request by nwalters512**
Following up on #220. This PR by no means converts fully to TypeScript, but it does add it to our toolchain so that we can incrementally add types in the future. The build process is a little wonky given that Next doesn't support transpiling server files itself (see https://github.com/zeit/next.js/issues/1735) so we have to work around that. I'm going to write docs explaining what I did and why I did it so that if problems come up in the future, future maintainers have something to go off of.

**TODO**

- [x] Document the build process

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -4,17 +4,18 @@
...
     "start": "NODE_ENV=production node build/server.js",
     "now-start": "NODE_ENV=now node build/server.js",
     "test": "NODE_ENV=test jest",
     "test-sequential": "jest --runInBand",
-    "lint": "eslint --ext .js,.jsx ./src",
-    "fix-lint": "npm run lint -- --fix",
-    "prettier": "prettier --write \"src/**/*.{js,jsx}\"",
-    "prettier-diff": "prettier --list-different \"src/**/*.{js,jsx}\"",
+    "lint-js": "eslint --ext .js,.jsx ./src",
+    "fix-lint-js": "npm run lint-js -- --fix",
```
Comment (james9909) [inline package.json]: Maybe we could add a `lint-ts` target for TypeScript linting with [typescript-eslint](https://github.com/typescript-eslint/typescript-eslint)?

File: src/util/index.ts [hunk 2]
```diff
@@ -1,25 +1,37 @@
 /* eslint-env browser */
 
-module.exports.baseUrl =
+declare global {
```
Comment (james9909) [inline src/util/index.ts]: We might want to create our own module with `.d.ts` files in the future, so we can centralize some of our types that may be shared across other modules.

Comment (nwalters512) [reply src/util/index.ts]: I'm going to punt this down the road a bit - agree that centralized types will be good (strongly-typed APIs? yes pls!) but we can make that happen as it's needed.

**Comment by nwalters512** [comment]
Did my best to add a suitable linting config for TypeScript - there's a bit of weird duplication of some rules between JS and TS and I had to add some overrides since there's not great built-in support for dual-language projects in `typescript-eslint`. Lmk how things look to you!

**Pull request merged**